met-web/src/components/engagement/view/PreviewBanner.tsx

```diff
@@ -10,50 +10,43 @@ export const PreviewBanner = () => {
     const navigate = useNavigate();
     const { engagementLoading, savedEngagement, publishEngagement } = useContext(ActionContext);
     const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const [isDraft, setIsDraft] = useState(true);
 
     const engagementId = savedEngagement.id || '';
-    if (engagementLoading) {
-        return <Skeleton variant="rectangular" width="100%" height="10em" />;
-    }
-    const isDraft = savedEngagement.status_id === EngagementStatus.Draft;
 
-    const validate = () => {
-        const errors = {
-            status: savedEngagement.status_id !== EngagementStatus.Draft,
-        };
-
-        return Object.values(errors).some((isError: unknown) => isError);
-    };
+    useEffect(() => {
+        setIsDraft(savedEngagement.status_id === EngagementStatus.Draft);
+    }, [savedEngagement]);
 
     const handlePublishEngagement = async () => {
-        const hasErrors = validate();
```

Comment:
can we return early instead of if else. if(!isDraft) navigate('/'); return;) then the rest of the code not under an if.  Less cognitive complexity this way

Reply:
Updated